conda/cli/main_run.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import json
+import os
+from os.path import isfile, join
+from subprocess import Popen
+import sys
+from tempfile import NamedTemporaryFile
+
+from .. import CondaError
+from ..base.context import context
+from ..common.compat import ensure_binary, iteritems, on_win
+from ..exceptions import CommandNotFoundError
+from ..gateways.disk.delete import rm_rf
+from ..gateways.subprocess import subprocess_call
+
+
+class ExecutableNotFound(CondaError):
+
+    def __init__(self, target_prefix, executable_name):
+        message = ("The executable was not found in the target prefix.\n"
+                   "  target prefix: %(target_prefix)s\n"
+                   "  executable name: %(executable_name)s"
+                   )
+        super(ExecutableNotFound, self).__init__(message, target_prefix=target_prefix,
+                                                 executable_name=executable_name)
+
+
+def _get_activated_env_vars_win(env_location):
+    try:
+        conda_bat = os.environ["CONDA_BAT"]
+    except KeyError:
+        raise CommandNotFoundError("run")
+
+    temp_path = None
+    try:
+        with NamedTemporaryFile('w+b', suffix='.bat', delete=False) as tf:
+            temp_path = tf.name
+            tf.write(ensure_binary(
+                "@%CONDA_PYTHON_EXE% -c \"import os, json; print(json.dumps(dict(os.environ)))\""
+            ))
+        cmd_builder = [
+            "cmd.exe /K \"",
+            "@SET PROMPT= ",
+            "&&",
+            "@SET CONDA_CHANGEPS1=false"
+            "&&",
+            "@CALL {0} activate \"{1}\"".format(conda_bat, env_location),
+            "&&",
+            "\"{0}\"".format(tf.name),
+            "\"",
+        ]
+        cmd = " ".join(cmd_builder)
+        result = subprocess_call(cmd)
+    finally:
+        if temp_path:
+            rm_rf(temp_path)
+
+    assert not result.stderr, result.stderr
+    env_var_map = json.loads(result.stdout)
+    return env_var_map
+
+
+def _get_activated_env_vars_unix(env_location):
+    try:
+        conda_exe = os.environ["CONDA_EXE"]
+    except KeyError:
+        raise CommandNotFoundError("run")
+
+    cmd_builder = [
+        "sh -c \'"
+        "eval \"$(\"{0}\" shell.posix hook)\"".format(conda_exe),
+        "&&",
+        "conda activate \"{0}\"".format(env_location),
+        "&&",
+        "\"$CONDA_PYTHON_EXE\" -c \"import os, json; print(json.dumps(dict(os.environ)))\"",
```

Comment:
Likewise, for Unix: why not put the user command in this line? We have the environment activated and ready to go. Let's use it!

Reply:
Haha makes sense.  This code was first pulled from https://github.com/tox-dev/tox-conda/pull/1, and I was planning on porting it back once we had this up and working.  If we do as you suggest, it's not as composable and reusable.  I don't mind doing as you suggest though, after we get this working for the other use case.